Improve PT and EN translations

eduardomozart opened the pull request:
This commit contains some bugfixes related to the language handling on Forma LMS and also attempts to add some missing strings on PT and EN languages files, and also attempts to updates the PT translation to meet the spelling agreement between Portugal and Brazil. Some commits contains a Description explaining what it fixes and they can be seen on "Commits" tab of this PR, clicking on the "..." (three periods) after each commit title.

I believe that the installer strings should be incorporated on XML_language.xml file instead of relying on PHP files. I also noticed that the XML language strings isn't uniform between the XML files, so some strings are missing and Forma LMS doesn't have any fallback mechanism to English or another language. (1) Maybe the English XML language should be installed by default and used as a fallback when a string doesn't exists, or (2) at least the English XML file should be updated when a new string is created (notice that the focus seems the Italian language), and a script could be created to parse the English XML file and update the other XML language files to include those missing strings (they will be shown in English on languages where it wasn't set, but it's better than it's variables names), or (3) (which seems more easily and reachable for me) is that the ``core_lang_text`` table from DB values could be imported from a reference XML file (e.g. Italian) and every imported translation would know which strings are missing and translate them. Could even be a tip on the Translation page instructing the users to contrib with their translations here on GitHub.

PS: I don't have money to become an associate to FormaLMS project, but it would be great if I had access to Forma LMS 4 considering my last contributions to the code (I was even cited on FormaLMS 3.2.22 Changelog). Thank you! =)

eduardomozart committed:
Add missing strings on English XML file

eduardomozart committed:
Improve PT translation

This commit adds missing PT tags, fix some texts and also attempts to standardize PT-BR and PT-PT (when possible, privileging PT-PT terms)

eduardomozart committed:
Improve installer and XML PT language, fix translation on last step

eduardomozart committed:
Update portuguese.php

eduardomozart committed:
Fix user status str

eduardomozart committed:
Improve title rendering on Language mgmt screen

eduardomozart committed:
Fix accents rendering

When the XML files uses accents and special chars (like cedilla "ç"), e.g. the "_DIFF_LANG" string from standard module, when hovering it on Language Management screen, the tooltip text isn't rendered as expected. From what I've seem, it seems to be related on how the translations were stored on XML language on the past, as it replaces accents with their HTML entity counterpart e.g. "é" was stored as "&egrave;" on PHP/XML language files. It seems that when exporting XML files right now, this conversion doesn't happen, so the "_DIFF_LANG" from Portuguese is stored as "Comparação" instead of it's HTML entity. If the XML export is wrong, it should be fixed, but if it's right, this fix make sure that the "Comparação" string is properly rendered on the page.

eduardomozart committed:
Fix MySQL str display

eduardomozart committed:
Add missing str on PT and EN XML language files

eduardomozart committed:
Remove unused str from PT translation

eduardomozart committed:
Add missing strings

eduardomozart committed:
Typo fixes and fix loading of translations edit screen

When searching for a text on DataTables and reloading the pages, the filtering was still being applied but the text isn't shown on text input. E.g. searching for "calend" on "Diff" field, it filters the results by the "calend" string as expected, but when reloading the page, it was still filtering the results by "calend" but the text isn't shown on "Diff" field. It's a known issue as can be seen here https://stackoverflow.com/questions/49878160/data-table-search-is-empty-after-refresh-but-filter-is-active and would require writing a custom initComplete function to repopulate the search text to the field, but as there's at least three search fields, I couldn't found a way to properly detect which of them has the search str and repopulate it, so I decided to disable the stateSave feature that fixes this behavior. Now every time the page is reloaded the page is loaded from scratch withoutr filtering.

eduardomozart committed:
Fix PT typos

eduardomozart committed:
Add Facebook oAuth translation strings

eduardomozart committed:
String update

formalms commented:
> PS: I don't have money to become an associate to FormaLMS project, but it would be great if I had access to Forma LMS 4 considering my last contributions to the code (I was even cited on FormaLMS 3.2.22 Changelog). Thank you! =)

@eduardomozart contact us to become a contributor, we will happy to onboard you

eduardomozart committed:
Fix API Token concat on PHP 8

Please see https://github.com/formalms/formalms/issues/22

eduardomozart commented:
> > PS: I don't have money to become an associate to FormaLMS project, but it would be great if I had access to Forma LMS 4 considering my last contributions to the code (I was even cited on FormaLMS 3.2.22 Changelog). Thank you! =)
> 
> @eduardomozart contact us to become a contributor, we will happy to onboard you

Hello, thank you! How could I do that?

eduardomozart committed:
Normalize certificate string tags substitions

When generating a certificate for a user with 'firstname' set as "André" (notice the accent on "e") on a MySQL UTF-8 database, it's saved as 65CC81 (Hex) on DB, which is "e" followed by "'". TCPDF supports custom fonts, and while using it's built-in fonts these characters combinations are rendered as expected, when using custom fonts (please see https://docs.moodle.org/404/en/Add_fonts_for_embedding), most TrueType fonts doesn't contains all UTF-8 grypths (e.g. Californian FB), so the [firstname] tag is rendered as "Andre?" instead of "André". Unicode provides various ways to represent the same character, which can lead to inconsistencies in text processing and comparison. Normalization aims to address this issue by converting text into a standardized form, ensuring that equivalent strings with different representations are treated as identical. After normalizing the string, it's output ("André") is similar to Latin1 (ISO-8859-1)/windows-2252 form and it's rendered as expected by custom TTF fonts.

eduardomozart committed:
Translate error message while editing user profile

If user attempts to rename the userid to another userid in use, the operation fails and an English message is shown. This fix creates a new variable to display the error message in current user locale.

eduardomozart committed:
Merge branch 'master' of https://github.com/eduardomozart/formalms

eduardomozart committed:
Add "Select all" key function to inputs on Translate table

When using the "Translation" and "Comparison" table fields, when pressing "Ctrl" + "A" on field, the input text isn't selected. This commits select all text on input when "Ctrl" + "A" is pressed on keyboard.

eduardomozart committed:
Update portuguese translation

eduardomozart committed:
Update lang[portuguese].xml

eduardomozart committed:
Update lang[portuguese].xml
